dist/tools/ci/can_fast_ci_run.py

```diff
@@ -0,0 +1,223 @@
+            self.__add_module(self.modules, file)
+        elif REGEX_PKG.match(file):
+            self.__add_module(self.pkgs, file)
+        elif REGEX_BOARD.match(file):
+            self.__add_module(self.boards, file)
+        elif REGEX_APP.match(file):
+            self.__add_module(self.apps, file)
+        else:
+            raise Exception("File \"{}\" doesn't match any known category".format(file))
+
+    def print_files_and_classifications(self):
+        """
+        Print all files and their classification in human readable format
+        """
+        print_change_set_section("Other", self.other)
+        print_change_set_section("Modules", self.modules)
+        print_change_set_section("Packages", self.pkgs)
+        print_change_set_section("Boards", self.boards)
+        print_change_set_section("Apps", self.apps)
+
+
+def classify_changes(riotbase=None, upstream_branch="master"):
+    """
+    Runs the given compiler with -v -E on an no-op compilation unit and parses the built-in
+    include search directories and the GCC version from the output
+
+    :param args: parse command line arguments
+    :type args: dict
+    :param pr_branch: name of the PR branch
+    :type pr_branch: str
+    :param upstream_branch: name of the main upstream branch the PR should be merged into
+    :type upstream_branch: str
+
+    :return: True if fast rebuilt is possible, False otherwise
+    :rtype: bool
+    """
+    change_set = ChangeSet(riotbase)
+
+    with subprocess.Popen(["git", "diff", "--numstat", f"HEAD..{upstream_branch}"],
+                          stdout=subprocess.PIPE) as proc:
+        for line in io.TextIOWrapper(proc.stdout, encoding="utf-8"):
+            match = REGEX_GIT_DIFF_RENAME_COMPLEX.match(line)
+            if match:
+                prefix = match.group(1)
+                suffix = match.group(4)
+                file_before = prefix + match.group(2) + suffix
+                file_after = prefix + match.group(3) + suffix
+                change_set.add_file(file_before)
+                change_set.add_file(file_after)
+                continue
+
+            match = REGEX_GIT_DIFF_RENAME_SIMPLE.match(line)
+            if match:
+                file_before = match.group(1)
+                file_after = match.group(2)
+                change_set.add_file(file_before)
+                change_set.add_file(file_after)
+                continue
+
+            match = REGEX_GIT_DIFF_SINGLE_FILE.match(line)
+            if match:
+                file = match.group(1)
+                change_set.add_file(file)
+                continue
+
+            raise Exception("Failed to parse \"{}\"".format(line))
+
+    return change_set
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Check if a fast CI run is possible and which "
+                                     + "boards / applications to build")
+    parser.add_argument("--explain", default=False, action="store_const", const=True,
```

Comment:
You could do this with the same result (here and below):

```suggestion
    parser.add_argument("--explain", action="store_true",
```